packages/spec-lib/src/decorators.ts

```diff
@@ -0,0 +1,246 @@
+import {
+  $service,
+  DecoratorContext,
+  Enum,
+  getNamespaceFullName,
+  getTypeName,
+  Interface,
+  listServices,
+  Model,
+  Namespace,
+  Operation,
+  Program,
+  StringLiteral,
+} from "@typespec/compiler";
+import {
+  $route,
+  $server,
+  getOperationVerb,
+  getRoutePath,
+  getServers,
+  HttpVerb,
+} from "@typespec/http";
+import { $versioned } from "@typespec/versioning";
+
+// Allow transition for breaking change https://github.com/microsoft/typespec/pull/1877
+function unifyString(type: string | StringLiteral) {
+  return typeof type === "string" ? type : type.value;
+}
+
+const ScenarioDocKey = Symbol("ScenarioDoc");
```

Comment:
Should use the State defined in `$lib`

Reply:
@timotheeguerin Can you please elaborate on the comment? I am not able to understand the feedback. Thanks

Reply:
See how xml library is defining its state keys https://github.com/microsoft/typespec/blob/d0666e643e56775d4fc7c53c814e37677530a80d/packages/xml/src/lib.ts#L39-L45

This is the current recommended way to go about it.

Reply:
Done